packages/core/lib/console.js

```diff
@@ -323,38 +320,47 @@ class Console extends EventEmitter {
       childPath = path.join(__dirname, "../lib/console-child.js");
     }
 
-    // stderr is piped here because we don't need to repeatedly see the parent
-    // errors/warnings in child process - specifically the error re: having
-    // multiple config files
-    const spawnOptions = { stdio: ["inherit", "inherit", "pipe"] };
+    const spawnOptions = { stdio: "pipe" };
     const settings = ["config", "network", "url"]
       .filter(setting => options[setting])
       .map(setting => `--${setting} ${options[setting]}`)
       .join(" ");
 
     const spawnInput = `${settings} -- ${inputStrings}`;
 
-    const spawnResult = spawnSync(
+    const sid = spawn(
       "node",
       ["--no-deprecation", childPath, spawnInput],
       spawnOptions
     );
 
-    if (spawnResult.stderr) {
-      // Theoretically stderr can contain multiple errors.
-      // So let's just print it instead of throwing through
-      // the error handling mechanism. Bad call?
-      debug(spawnResult.stderr.toString());
-    }
+    // Theoretically stderr can contain multiple errors.
+    // So let's jsut print it instaed of throwing through
+    // the error handling mechanism. Bad call?
+    sid.stderr.on("data", data => {
+      debug(data.toString());
```

Comment:
IIRC, stderr's `data` event might emit in chunks; `data` might be in the middle of a line. Here is a StackOverflow question about it: https://stackoverflow.com/questions/18243810/handling-chunked-responses-from-process-stdout-data-event

Reply:
@gnidan Any thoughts on this corner case? I don't think `console-child.js` is actually capable of causing the above issue, so it's probably fine to ignore this case.

Reply:
Oh yeah definitely should make sure we buffer this data!

Reply:
Additional context. The data is from stderr and was previously fed to debug, a conditional, decorated console.error. Someone would have to invoke `DEBUG=console truffle develop` to hook into it.